picky-asn1-x509/src/subject_public_key_info.rs

```diff
@@ -42,10 +46,10 @@ impl SubjectPublicKeyInfo {
         }
     }
 
-    pub fn new_ec_key<P: Into<BitStringAsn1>>(ec_point: P) -> Self {
+    pub fn new_ec_key(curve: ObjectIdentifier, point: BitString) -> Self {
         Self {
-            algorithm: AlgorithmIdentifier::new_elliptic_curve(EcParameters::NamedCurve(oids::ec_public_key().into())),
```

Comment:
That was a bug, which caused OpenSSL to freakout - writing invalid OID for a curve (we were writing not a curve oid at all